### ISSUE TITLE
Improve string ergonomics

### ISSUE THREAD

**Issue by x87** [sannybuilder/dev#255]
- [x] string literals (both single and double quoted) should support escape sequences. currently only strings inside hex..end do that:
```
hex
  "\t" // compiles as 09 (code for a TAB character)
end

0@v = "\t" // compiles as 5C 74 ("\t")
```

- [x] strings should support multiple spaces without [escaping](https://docs.sannybuilder.com/coding/hex..end#escape-sequences)

```
hex
  "\t     \t" // leave one space between \t
end

0@v = "\t     \t" // same
```

- [x] SB3_compat mode

**Comment by x87** [comment]
Need compatibility mode to allow old scripts compile without having to make changes to strings using \{char} (e.g. "data\test.cs"

**Comment by x87** [comment]
released in 4.0.0
